app/routes/circulars._archive._index.tsx

```diff
@@ -239,10 +243,55 @@ export default function () {
   if (searchString) searchString = `?${searchString}`
 
   const [inputQuery, setInputQuery] = useState(query)
-  const clean = inputQuery === query
+  const [inputDateGte, setInputDateGte] = useState(startDate)
+  const [inputDateLte, setInputDateLte] = useState(endDate)
+  const [showContent, setShowContent] = useState(false)
+  const [showDateRange, setShowDateRange] = useState(false)
+  const clean =
+    inputQuery === query &&
+    inputDateGte === startDate &&
+    inputDateLte === endDate
 
   const submit = useSubmit()
 
+  const setFuzzyTime = (startDate?: string) => {
+    if (startDate === 'custom') {
+      setShowDateRange(true)
+    } else {
+      setShowDateRange(false)
+      setInputDateGte(startDate)
+    }
+    setInputDateLte('now')
+  }
+
+  const setDateRange = () => {
+    setShowContent(false)
+    const params = new URLSearchParams(location.search)
+    if (inputDateGte) params.set('startDate', inputDateGte)
+    if (inputDateLte) params.set('endDate', inputDateLte)
+    submit(params, {
+      method: 'get',
+      action: '/circulars',
+    })
+  }
+
+  useEffect(() => {
+    const radiobuttons = document.querySelectorAll('input[name="radio-date"]')
+    let radioSelected = false
+    radiobuttons.forEach((button) => {
+      if ((button as HTMLInputElement).value === inputDateGte) {
+        ;(button as HTMLInputElement).checked = true
+        radioSelected = true
+      }
+    })
+    if (!radioSelected) {
+      const customButton = document.getElementById('radio-custom')
+      if (customButton) {
+        ;(customButton as HTMLInputElement).checked = true
+      } // does not address the need for the custom date range button to be expanded if this button is checked
+    }
+  })
```

Comment:
No, don't use `useEffect` here. You can generate the value for `defaultChecked` from the search params.

Reply:
I had thought defaultChecked was just the way to specify with radio to have selected by default, is that not the case?

I had tried doing something like `checked={isChecked(buttonValue)}` for each of the radio buttons, where `isChecked()` passed the specific button value to a function and checked if it was the same as the search param. This worked for having it show the correct radio button as selected after submitting, but it wouldn't update to reflect choosing a new radio button prior to submission. 

In this example, the page has been loaded with the radio button selected for the last day. Upon trying to click on the option for the last year, it shows that the radio has been selected, but the last year button doesn't actually show as being checked until you submit and reload the page:
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/57914086/3a4a627d-9264-4bdf-babb-cfba89c5be7c)

Reply:
> I had thought defaultChecked was just the way to specify with radio to have selected by default, is that not the case?

Correct.

> I had tried doing something like `checked={isChecked(buttonValue)}` for each of the radio buttons, where `isChecked()` passed the specific button value to a function and checked if it was the same as the search param. This worked for having it show the correct radio button as selected after submitting, but it wouldn't update to reflect choosing a new radio button prior to submission.

You generally should _not_ explicitly set the value or the checked state of controls that the user is interacting with.